controllers/slicegateway/slicegateway.go

```diff
@@ -822,3 +873,76 @@ func contains(s []string, e string) bool {
 	}
 	return false
 }
+
+// total -> external ip list of nodes in the k8s cluster
+// current -> ip list present in nodeIPs of cluster cr
+func validatenodeipcount(total, current []string) bool {
+	for _, a := range total {
+		if !contains(current, a) {
+			return false
+		}
+	}
+	return true && len(total) == len(current)
+}
+func UpdateGWPodStatus(gwPodStatus []*kubeslicev1beta1.GwPodInfo, podName string) []*kubeslicev1beta1.GwPodInfo {
+	index := -1
+	for i, _ := range gwPodStatus {
+		if gwPodStatus[i].PodName == podName {
+			index = i
+			break
+		}
+	}
+	return append(gwPodStatus[:index], gwPodStatus[index+1:]...)
+
+}
+func getLocalNSMIPs(slicegateway *kubeslicev1beta1.SliceGateway) []string {
+	nsmIPs := make([]string, 0)
+	for i, _ := range slicegateway.Status.GatewayPodStatus {
+		nsmIPs = append(nsmIPs, slicegateway.Status.GatewayPodStatus[i].LocalNsmIP)
+	}
+	return nsmIPs
+}
+func getPodIPs(slicegateway *kubeslicev1beta1.SliceGateway) []string {
+	podIPs := make([]string, 0)
+	for i, _ := range slicegateway.Status.GatewayPodStatus {
+		podIPs = append(podIPs, slicegateway.Status.GatewayPodStatus[i].PodIP)
+	}
+	return podIPs
+}
+func getPodNames(slicegateway *kubeslicev1beta1.SliceGateway) []string {
+	podNames := make([]string, 0)
+	for i, _ := range slicegateway.Status.GatewayPodStatus {
+		podNames = append(podNames, slicegateway.Status.GatewayPodStatus[i].PodName)
+	}
+	return podNames
+}
+func isGWPodStatusChanged(slicegateway *kubeslicev1beta1.SliceGateway, gwPod *kubeslicev1beta1.GwPodInfo) bool {
+	gwPodStatus := slicegateway.Status.GatewayPodStatus
+	for i, _ := range gwPodStatus {
+		if gwPodStatus[i].PodName == gwPod.PodName {
+			return gwPodStatus[i].TunnelStatus.Status == gwPod.TunnelStatus.Status
+		}
+	}
+	return false
+}
+func getPodAntiAffinity(slice string) *corev1.PodAntiAffinity {
+	return &corev1.PodAntiAffinity{
+		PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{{
+			Weight: 1,
+			PodAffinityTerm: corev1.PodAffinityTerm{
+				TopologyKey: controllers.TopologyKeySelector,
+				LabelSelector: &metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{{
+						Key:      controllers.PodTypeSelectorLabelKey,
+						Operator: metav1.LabelSelectorOpIn,
+						Values:   []string{"slicegateway"},
+					}, {
+						Key:      controllers.ApplicationNamespaceSelectorLabelKey,
```

Comment:
Please confirm that this label is added to the gw pods when the gw deployment is created.